### PR TITLE
Use Scala futures

### PR DESCRIPTION
**Pull request by alexbergeron**

### CODE REVIEW

File: mry-core/src/test/scala/com/wajam/mry/TestDatabase.scala [hunk 1]
```diff
@@ -52,12 +62,13 @@ class TestDatabase extends FunSuite with BeforeAndAfterAll {
         b.from("memory").set(key, "value%s".format(key))
       })
 
+
       db.execute(b => {
         val context = b.from("context")
         b.returns(b.from("memory").get(key), context.get("tokens"), context.get("local_node"))
-      }, p.complete(_, _))
+      }, complete(p))
```
Comment (coderunner) [inline mry-core/src/test/scala/com/wajam/mry/TestDatabase.scala]: Shouldn't the test use the execute methods that returns a Future?

Comment (alexbergeron) [inline mry-core/src/test/scala/com/wajam/mry/TestDatabase.scala]: +1

**Comment by alexbergeron** [comment]
Updated

**Comment by coderunner** [comment]
+1

**Pull request merged**